lib/Mojo/Base.pm

```diff
@@ -82,9 +81,16 @@ sub import {
 
   # "has" and possibly ISA
   if ($flags[0]) {
-    no strict 'refs';
-    push @{"${caller}::ISA"}, $flags[0] unless $flags[0] eq '-role';
     Mojo::Util::monkey_patch($caller, 'has', sub { attr($caller, @_) });
+    if ($flags[0] eq '-role') {
+
+      # import of Role::Tiny must come after injecting has
+      eval "package $caller; use Role::Tiny; 1" or die $@;
+    }
+    else {
+      no strict 'refs';
+      push @{"${caller}::ISA"}, $flags[0] unless $flags[0] eq '-role';
+    }
```

Comment:
The unless here seems superflous since the unless already indicates the lack of '-role'?